#!/usr/bin/env python

PACKAGE = "dji_m100_trajectory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name			Type      	Level	Description			Default Min	Max
gen.add(	"Llidar_nan",		bool_t,		0,	"nan values for Llidar",	False)
gen.add( 	"Llidar_range",		double_t, 	0, 	"range",  			0.707, 	0, 	5)
gen.add( 	"Llidar_min_range",	double_t, 	0, 	"min_range",  			0.5, 	0, 	5)
gen.add( 	"Llidar_max_range",	double_t, 	0, 	"max_range",  			5, 	0, 	5)
gen.add(	"Clidar_nan",		bool_t,		0,	"nan values for Clidar",	False)
gen.add( 	"Clidar_range",		double_t, 	0, 	"range",  			1, 	0, 	5)
gen.add( 	"Clidar_min_range",	double_t, 	0, 	"min_range",  			0.5, 	0, 	5)
gen.add( 	"Clidar_max_range",	double_t, 	0, 	"max_range",  			5, 	0, 	5)
gen.add(	"Rlidar_nan",		bool_t,		0,	"nan values for Rlidar",	False)
gen.add( 	"Rlidar_range",		double_t, 	0, 	"range",  			0.707, 	0, 	5)
gen.add( 	"Rlidar_min_range",	double_t, 	0, 	"min_range",  			0.5, 	0, 	5)
gen.add( 	"Rlidar_max_range",	double_t, 	0, 	"max_range",  			5, 	0, 	5)
gen.add(	"sonar_nan",		bool_t,		0,	"nan values for sonar",	False)
gen.add( 	"sonar_range",		double_t, 	0, 	"range",  			1, 	0, 	5)
gen.add( 	"sonar_min_range",	double_t, 	0, 	"min_range",  			0.5, 	0, 	5)
gen.add( 	"sonar_max_range",	double_t, 	0, 	"max_range",  			5, 	0, 	5)

exit(gen.generate(PACKAGE, "dummyLidar", "set_dummyLidar"))
